src/picongpu/include/particles/ionization/byField/BSI/AlgorithmBSI.hpp

```diff
@@ -53,11 +54,13 @@ namespace ionization
         operator()( const BType bField, const EType eField, ParticleType& parentIon )
         {
 
+            const float_X protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
+            
             /* ionization condition */
-            if (math::abs(eField)*UNIT_EFIELD >= SI::IONIZATION_EFIELD && parentIon[chargeState_] < SI::PROTON_NUMBER)
+            if (math::abs(eField)*UNIT_EFIELD >= SI::IONIZATION_EFIELD && parentIon[chargeState_] < protonNumber)
             {
                 /* set new particle charge state */
-                parentIon[chargeState_] = 1 + parentIon[chargeState_];
+                parentIon[chargeState_] = float_X(1.0) + parentIon[chargeState_];
```

Comment:
probably better suited and easier to read:
```c++
parentIon[chargeState_] += float_X(1.0);
```
(sry, didn't see that the last time)